### PR TITLE
Support scans on indices

### PR DESCRIPTION
**Pull request by majelbstoat**
Hello @emmazhou, @jimmyoneill, @xiao, 

Please review the following commits I made in branch 'jamie/scan-index'.

8c787960ad32e81fdb0aafc50e38cbd7815b6f6e (2017-04-04 18:47:02 -0700)
Update version and contributors

9a3473b618004acecd0763ce24dce0886e30f211 (2017-04-04 18:44:25 -0700)
Test parallel scan on index

68bb97e5a81790c32f32864b5c168b245baaa533 (2017-04-04 18:31:37 -0700)
Support scans on indices

R=@emmazhou
R=@jimmyoneill
R=@xiao

### CODE REVIEW

File: lib/ScanBuilder.js [hunk 1]
```diff
@@ -13,6 +14,19 @@ function ScanBuilder(options) {
 }
 require('util').inherits(ScanBuilder, Builder)
 
+/**
+ * If this query runs on a local index or global index, set a
```
Comment (xiao) [inline lib/ScanBuilder.js]: s/query/scan

File: test/utils/testUtils.js [hunk 2]
```diff
@@ -75,12 +75,24 @@ utils.deleteTable = function (db, tableName) {
...
+}
+
+/*
  * A helper function that creates the testing table.
  *
  * @param db {AWS.DynamoDB} The database instance.
  * @return {Promise}
  */
-utils.createTable = function (db, tableName, hashKey, rangeKey) {
+utils.createTable = function (db, tableName, hashKey, rangeKey, gsiDefinitions) {
```
Comment (xiao) [inline test/utils/testUtils.js]: Nice! While you're here, could you also implement LSI and test SCAN on LSIs as well?

File: test/testScan.js [hunk 3]
```diff
@@ -62,6 +62,29 @@ builder.add(function testScanSegment(test) {
...
+               .setHashKey('description')
+  return scanAndCheck(scan, [5], test)
+})
+
+builder.add(function testParallelScanOnGlobalSecondaryIndex(test) {
+  var scan = this.client.newScanBuilder(tableName)
+               .setIndexNameGenerator(utils.indexNameGenerator)
+               .setHashKey('post')
+               .setParallelScan(1, 2)
+  return scanAndCheck(scan, [0, 2, 5, 6], test)
```
Comment (xiao) [inline test/testScan.js]: Is this deterministic? I mean it will always be [0, 2, 5, 6], but never be [0, 2, 3, 4] or other combinations?

Comment (majelbstoat) [inline test/testScan.js]: i'm pretty confident this is deterministic, yes. if it wasn't, the segmentation strategy that amazon used could lead to overlaps or missed rows.

File: test/utils/testUtils.js (hunk 2 above)
Comment (majelbstoat) [inline test/utils/testUtils.js]: i will, but i'll do it as a follow up if that's ok. we need the GSI support for a product feature in the short term.

Comment (xiao) [inline test/utils/testUtils.js]: Cool. Thanks!

**Pull request merged**